lib/api/api_auth.go

```diff
@@ -166,6 +166,31 @@ func authLDAP(username string, password string, cfg config.LDAPConfiguration) bo
 		return false
 	}
 
+	if cfg.SearchFilter != "" {
```

Comment:
Should check that base is not empty as well as otherwise it will make no sense afaik

Reply:
I changed this slightly but the new logic is the same - if either is set we run the code. It'll fail when just one is set, but I'd rather we fail always if just the filter is set than let authentication succeed and the filter just *look* active

Reply:
I guess we could check for it specifically and fail the auth with a specific error pointing out the config problem...